framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1306,6 +1324,26 @@ protected void postDirectSuperTypes(
     public void postAsMemberOf(
             AnnotatedTypeMirror type, AnnotatedTypeMirror owner, Element element) {
         addComputedTypeAnnotations(element, type);
+        if (withEffectiveViewpointAdaptation) {
+            if (viewpointAdaptor.shouldNotBeAdapted(type, element)) return;
+            AnnotatedTypeMirror decltype = this.getAnnotatedType(element);
+            AnnotatedTypeMirror combinedType =
+                    this.viewpointAdaptor.combineTypeWithType(owner, decltype, this);
+            // TODO Remove this ugly code
+            // Update: we don't have a method to do this job. I think we need to add this logic.
+            type.replaceAnnotation(viewpointAdaptor.getAnnotationMirror(combinedType, this));
```

Comment:
@wmdietl Hi prof, I searched through many possible candidate utility classes but didn't find any method that performs this replace operation on ADT and AAT. This is the last set of change you requested. Should I add this method to AnnotatedTypes? Or can you maybe start 3rd round of code review? Thank you!